### PR TITLE
BLS-aggregation integration tests

### PR DESCRIPTION
**Pull request by TomasArrachea**
Implemented integration tests for `bls_aggregation` crate.
Added 5 tests:

- 2 quorums 1 operator (ignored)
- 2 quorums 2 operators shared (ignored)
- 1 quorum 1 operator
- 1 quorum 2 operators 
- 2 quorums 2 operators separated

Two cases are ignored as they are failing due to this bug: [bls aggregation service doesn't support >1 quorum](https://github.com/Layr-Labs/eigensdk-go/issues/261)

### CODE REVIEW

File: crates/crypto/bls/src/lib.rs [hunk 1]
```diff
@@ -142,29 +189,79 @@ pub fn convert_to_g2_point(g2: G2Affine) -> Result<G2Point, BlsError> {
...
+}
 
-        Ok(G2Point {
-            X: [x_u256_1, x_u256_0],
-            Y: [y_u256_1, y_u256_0],
-        })
-    } else {
-        Err(BlsError::InvalidG2Affine)
-    }
+/// Convert [`G2Affine`] to [`G2Point`]
```
Comment (supernovahs) [inline crates/crypto/bls/src/lib.rs:238]:
```suggestion
/// Convert [`G2Affine`] to [`G2PointRegistry `]
```

File: crates/crypto/bls/src/lib.rs [hunk 2]
```diff
@@ -142,29 +189,79 @@ pub fn convert_to_g2_point(g2: G2Affine) -> Result<G2Point, BlsError> {
...
+    let y_point = g1_point.Y.into_limbs();
+    let y = Fq::new(BigInteger256::new(y_point));
+    G1Affine::new(x, y)
+}
 
-        let x_u256_0 = U256::from_limbs(x_0.0);
-        let x_u256_1 = U256::from_limbs(x_1.0);
-        let y_u256_0 = U256::from_limbs(y_0.0);
-        let y_u256_1 = U256::from_limbs(y_1.0);
+/// Convert [`G1Point`] to [`G1Affine`]
```
Comment (supernovahs) [inline crates/crypto/bls/src/lib.rs:225]: comment

File: crates/crypto/bls/src/lib.rs [hunk 3]
```diff
@@ -118,20 +116,69 @@ pub fn convert_to_g1_point(g1: G1Affine) -> Result<G1Point, BlsError> {
...
+    let x_u256 = U256::from_limbs(x.0);
+    let y_u256 = U256::from_limbs(y.0);
+
+    Ok(G1PointChecker {
+        X: x_u256,
+        Y: y_u256,
+    })
+}
+
+/// Convert [`G1Affine`] to  Alloy [`G1Point`]
```
Comment (supernovahs) [inline crates/crypto/bls/src/lib.rs:155]: comment update needed

File: crates/crypto/bls/src/lib.rs [hunk 4]
```diff
@@ -118,20 +116,69 @@ pub fn convert_to_g1_point(g1: G1Affine) -> Result<G1Point, BlsError> {
...
+    let x_u256 = U256::from_limbs(x.0);
+    let y_u256 = U256::from_limbs(y.0);
+
+    Ok(G1Point {
+        X: x_u256,
+        Y: y_u256,
+    })
+}
+
+/// Convert [`G1Affine`] to  Alloy [`G1Point`]
```
Comment (supernovahs) [inline crates/crypto/bls/src/lib.rs:135]:
```suggestion
/// Convert [`G1Affine`] to  Alloy [`G1PointChecker `]
```

**Review by supernovahs — CHANGES_REQUESTED** [review]

**Comment by supernovahs** [comment]
regarding the issue of adding separate conversion for different bindings. 
Maybe we can reuse the original conversion functions, instead of adding separate for each bindings by unwrapping the X and Y values and wrapping in appropriate binding variant in the crate we need. (we can also use the above conversion functions in the lib.rs of those crates and use those their as helper also)

If you think above is a better way . Otherwise all good. some small nits

**Comment by TomasArrachea** [comment]
> regarding the issue of adding separate conversion for different bindings. Maybe we can reuse the original conversion functions, instead of adding separate for each bindings by unwrapping the X and Y values and wrapping in appropriate binding variant in the crate we need. (we can also use the above conversion functions in the lib.rs of those crates and use those their as helper also)
> 
> If you think above is a better way . Otherwise all good. some small nits

@supernovahs I think it's better to merge the PR as is, and then make the refactor in a separate PR if needed.

**Comment by supernovahs** [comment]
needs a main merge

**Pull request merged**